src/Interpreters/Cluster.cpp

```diff
@@ -423,6 +423,18 @@ Cluster::Cluster(const Poco::Util::AbstractConfiguration & config,
     if (config_keys.empty())
         throw Exception(ErrorCodes::SHARD_HAS_NO_CONNECTIONS, "No cluster elements (shard, node) specified in config at path {}", config_prefix);
 
+
+    size_t keys_with_static_shard_number = std::ranges::count_if(config_keys.begin(), config_keys.end(), [& config, & config_prefix](const String& key)
```

Comment:
AFAIS, these checks are missed
- shard number should be 1-based
- shard numbers should be unique in cluster

As well as a test for such cases

Reply:
Also, shard_num internally is `UInt32`, so boundary checks are necessary as well. But first, I'd rather answer the [question](https://github.com/ClickHouse/ClickHouse/pull/72276#pullrequestreview-2467380577) - is it worth the hassle?